### PR TITLE
subscribe to subset messages from glue

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This draft pull request attempts to subscribe to subset messages from glue (creation or update) and print a debugging message (eventually to be replaced by overriding subset thickness, etc).  However, the glue messages don't seem to be sent when creating a subset region in jdaviz (in specviz).

Open specviz example notebook, create a new subset.  Print message would show if the messages from glue were being sent and appropriately received, but currently are not.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by kecnry** [comment]
One workaround that I found that could work for our purposes here (subset thickness) is to instead hook into the callback created by brushing.  The downside is this hook needs to be initialized per-select-tool and will require extra logic to determine whether it created a new subset (vs if modifying an existing subset) and then retrieve the new subset to change the defaults.

File: jdaviz/configs/specviz/helper.py [hunk 1]
```diff
@@ -23,6 +25,18 @@ def __init__(self, *args, **kwargs):
         self.app.hub.subscribe(self, RedshiftMessage,
                                handler=self._redshift_listener)
 
+        self.app.session.hub.subscribe(self, SubsetMessage,
```
Comment (astrofrog) [inline jdaviz/configs/specviz/helper.py:28]: Why does this use ``app.session`` instead of just ``app``? (which is different from the redshift message)

**Comment by astrofrog** [comment]
I've checked this and it does seem to work - it's just that the print statements aren't being shown, I think because glue-jupyter tools silence the default stdout and redirects it to a dedicated output widget. Try adding a cell with:

```
specviz.app.get_viewer_by_id('specviz-0').output_widget
```

and then make the selection and you should then see:

```
*** _on_subset_msg SubsetCreateMessage: 
	 Sent from: Subset: Subset 1 (data: myfile)
*** _on_subset_msg SubsetUpdateMessage: 
	 Sent from: Subset: Subset 1 (data: myfile)
	 Updated subset_state
```

I keep getting caught out by this so I think we might want to reconsider the whole stdout redirection.

**Comment by astrofrog** [comment]
You can also check this another way by writing the debug statements to a file instead of printing them (I tried this and it does work)

**Pull request closed without merging**